nixos/modules/tasks/filesystems/zfs.nix

```diff
@@ -616,6 +623,10 @@ in
               fi
               poolImported "${pool}" || poolImport "${pool}"  # Try one last time, e.g. to import a degraded pool.
             fi
+
+            ${optionalString (elem pool (map (e: elemAt (splitString "/" e) 0) clevisDatasets))
```

Comment:
similarly, this `optionalString` seems redundant, because when the condition is false, the string would have been an empty string anyway.

Reply:
Agreed